src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -495,11 +495,19 @@ internal static class Encoding
 
         internal static class MemoryStream
         {
+            internal delegate bool TryGetBufferDelegate(System.IO.MemoryStream stream, out ArraySegment<byte> buffer);
+
             internal static readonly Type Type = typeof(System.IO.MemoryStream);
 
-            internal static readonly MethodInfo GetBuffer = Type
+            internal static readonly Func<System.IO.MemoryStream, byte[]> GetBuffer = Type
+                .GetTypeInfo()
+                .GetDeclaredMethod(nameof(GetBuffer))
+                .CreateDelegate<Func<System.IO.MemoryStream, byte[]>>();
+
+            internal static readonly TryGetBufferDelegate TryGetBuffer = Type
                 .GetTypeInfo()
-                .GetDeclaredMethod(nameof(GetBuffer));
+                .GetDeclaredMethod(nameof(TryGetBuffer))
+                .CreateDelegate<TryGetBufferDelegate>();
```

Comment:
@nguerrera It shouldn't. [`CreateDelegate<T>`](https://github.com/svick/roslyn/blob/b6984cb7/src/Compilers/Core/Portable/InternalUtilities/ReflectionUtilities.cs#L104-L112) is an extension method that handles `null`.